### PR TITLE
Don't run shellcheck for rpm .spec files

### LINKED ISSUE #9: flymake-shellcheck shouldn't activate when editing RPM .spec files
I have added `flymake-shellcheck-load` as a `sh-mode-hook` as recommended in the documentation, but that mode is loaded for RPM .spec files as well, and they are not supported by shellcheck.

I've changed my configuration to the following:

```
(add-hook 'sh-mode-hook (lambda ()
                            (unless (eq sh-shell 'rpm)
                              (flymake-shellcheck-load))))
```

but perhaps this check should be done by `flymake-shellcheck-load`, or at least documented in the README.

### PR DESCRIPTION
**Pull request by 1player**
Fixes #9 

This is my first time contributing elisp, please feel free to suggest how to better rewrite this change.

### CODE REVIEW

File: flymake-shellcheck.el [hunk 1]
```diff
@@ -117,10 +117,13 @@ Check for problems, then call REPORT-FN with results."
         (process-send-region flymake-shellcheck--proc (point-min) (point-max))
         (process-send-eof flymake-shellcheck--proc)))))
 
+(defvar sh-shell)
```
Comment (federicotdn) [inline flymake-shellcheck.el:120]: Why do we need this declaration here?

**Review by federicotdn — CHANGES_REQUESTED** [review]
Thanks for opening this PR!

File: flymake-shellcheck.el (hunk 1 above)
Comment (1player) [reply flymake-shellcheck.el:120]: IIRC, flymake complains about `sh-shell` being a free variable

**Comment by federicotdn** [comment]
I'm not taking PRs for this project anymore, as this functionality is now built-in into Emacs 29.

**Pull request closed without merging**